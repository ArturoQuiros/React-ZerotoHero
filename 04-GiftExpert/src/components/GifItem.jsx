import propTypes from "prop-types";

const GifItem = ({ url, title }) => {
  return (
    <>
      <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
      </div>
    </>
  );
};

GifItem.propTypes = {
  title: propTypes.string.isRequired,
  url: propTypes.string.isRequired,
};
export default GifItem;
