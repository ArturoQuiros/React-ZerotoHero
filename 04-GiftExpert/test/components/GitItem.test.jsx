import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

const title = "";
const url = "";

describe("Tests on GifItem", () => {
  test("Match to Snapshot?", () => {
    const { container } = render(<GifItem title={title} url={url}></GifItem>);
    expect(container).toMatchSnapshot();
  });

  test("must show the img with the URL", () => {
    render(<GifItem title={title} url={url}></GifItem>);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });
});
