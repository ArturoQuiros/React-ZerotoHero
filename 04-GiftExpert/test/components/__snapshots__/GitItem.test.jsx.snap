// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifItem Match to Snapshot? 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt=""
      src=""
    />
    <p />
  </div>
</div>
`;
